### ISSUE TITLE
arangodb-3.1.13 does not build on macOS Yosemite

### ISSUE THREAD

**Issue by fxcoudert** [arangodb/arangodb#2362]
arangodb-3.1.13 does not build on macOS Yosemite. The [pull request](https://github.com/Homebrew/homebrew-core/pull/10681) by your bot @arangodb-release-bot failed to build with the following error:

```
[ 92%] Building CXX object arangod/CMakeFiles/arangod.dir/Scheduler/SchedulerFeature.cpp.o
cd /tmp/arangodb-20170307-11482-x3rq7c/ArangoDB-3.1.13/build/arangod && /usr/local/Homebrew/Library/Homebrew/shims/super/clang++   -DARANGODB_BITS=64 -DARANGODB_BOOST_VERSION=\"1.62.0\" -DARANGODB_OPENSSL_VERSION=\"1.0.2k\" -DARANGODB_ROCKSDB_VERSION=\"\" -DARANGODB_V8_VERSION=\"5.0.71.39\" -DARANGODB_ZLIB_VERSION=\"1.2.8\" -DASM_OPTIMIZATIONS=0 -DBOOST_ALL_NO_LIB=1 -DCURL_STATICLIB=1 -DVELOCYPACK_XXHASH=1 -I/tmp/arangodb-20170307-11482-x3rq7c/ArangoDB-3.1.13/3rdParty/linenoise-ng/include -I/tmp/arangodb-20170307-11482-x3rq7c/ArangoDB-3.1.13/3rdParty/linenoise-ng/src -I/tmp/arangodb-20170307-11482-x3rq7c/ArangoDB-3.1.13/3rdParty/velocypack/include -I/tmp/arangodb-20170307-11482-x3rq7c/ArangoDB-3.1.13/3rdParty/rocksdb/rocksdb/include -I/tmp/arangodb-20170307-11482-x3rq7c/ArangoDB-3.1.13/build -I/tmp/arangodb-20170307-11482-x3rq7c/ArangoDB-3.1.13/build/lib -I/tmp/arangodb-20170307-11482-x3rq7c/ArangoDB-3.1.13 -I/tmp/arangodb-20170307-11482-x3rq7c/ArangoDB-3.1.13/arangod -I/tmp/arangodb-20170307-11482-x3rq7c/ArangoDB-3.1.13/arangosh -I/tmp/arangodb-20170307-11482-x3rq7c/ArangoDB-3.1.13/lib -I/tmp/arangodb-20170307-11482-x3rq7c/ArangoDB-3.1.13/3rdParty/V8/V8-5.0.71.39/third_party/icu/source/common -I/tmp/arangodb-20170307-11482-x3rq7c/ArangoDB-3.1.13/3rdParty/V8/V8-5.0.71.39/third_party/icu/source/i18n -I/tmp/arangodb-20170307-11482-x3rq7c/ArangoDB-3.1.13/3rdParty/V8/V8-5.0.71.39/third_party/icu/source/io -I/usr/local/opt/openssl/include -I/tmp/arangodb-20170307-11482-x3rq7c/ArangoDB-3.1.13/3rdParty/V8/V8-5.0.71.39/include -I/tmp/arangodb-20170307-11482-x3rq7c/ArangoDB-3.1.13/3rdParty/V8/V8-5.0.71.39 -I/tmp/arangodb-20170307-11482-x3rq7c/ArangoDB-3.1.13/3rdParty/zlib/zlib-1.2.8 -I/tmp/arangodb-20170307-11482-x3rq7c/ArangoDB-3.1.13/build/3rdParty/zlib/zlib-1.2.8 -I/tmp/arangodb-20170307-11482-x3rq7c/ArangoDB-3.1.13/3rdParty/curl/curl-7.50.3/include -I/tmp/arangodb-20170307-11482-x3rq7c/ArangoDB-3.1.13/build/3rdParty/curl/curl-7.50.3/include/curl -I/tmp/arangodb-20170307-11482-x3rq7c/ArangoDB-3.1.13/arangod/. -I/tmp/arangodb-20170307-11482-x3rq7c/ArangoDB-3.1.13/3rdParty/boost/SYSTEM -I/tmp/arangodb-20170307-11482-x3rq7c/ArangoDB-3.1.13/3rdParty/boost/1.62.0  -g                -Wall -Wextra -Wno-unused-parameter     -std=c++11 -stdlib=libc++ -O3 -fomit-frame-pointer        -Wall -Wextra -Wno-unused-parameter     -std=c++11 -stdlib=libc++ -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.11.sdk -mmacosx-version-min=10.11   -Wno-deprecated-declarations -std=gnu++11 -o CMakeFiles/arangod.dir/Scheduler/SchedulerFeature.cpp.o -c /tmp/arangodb-20170307-11482-x3rq7c/ArangoDB-3.1.13/arangod/Scheduler/SchedulerFeature.cpp
In file included from /tmp/arangodb-20170307-11482-x3rq7c/ArangoDB-3.1.13/arangod/Scheduler/JobQueue.cpp:23:
In file included from /tmp/arangodb-20170307-11482-x3rq7c/ArangoDB-3.1.13/arangod/Scheduler/JobQueue.h:26:
In file included from /tmp/arangodb-20170307-11482-x3rq7c/ArangoDB-3.1.13/lib/Basics/Common.h:154:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/algorithm:628:
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/memory:4032:35: error: no viable overloaded '='
                __e->__weak_this_ = *this;
                ~~~~~~~~~~~~~~~~~ ^ ~~~~~
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/memory:4070:5: note: in instantiation of function template specialization 'std::__1::shared_ptr<arangodb::Thread>::__enable_weak_this<(anonymous namespace)::JobQueueThread>' requested here
    __enable_weak_this(__p);
    ^
/tmp/arangodb-20170307-11482-x3rq7c/ArangoDB-3.1.13/arangod/Scheduler/JobQueue.cpp:112:7: note: in instantiation of function template specialization 'std::__1::shared_ptr<arangodb::Thread>::shared_ptr<(anonymous namespace)::JobQueueThread>' requested here
      _queueThread(new JobQueueThread(this, scheduler)) {}
      ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/memory:4984:15: note: candidate function not viable: no known conversion from 'std::__1::shared_ptr<arangodb::Thread>' to 'const std::__1::weak_ptr<(anonymous namespace)::JobQueueThread>' for 1st argument
    weak_ptr& operator=(weak_ptr const& __r) _NOEXCEPT;
              ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/memory:4995:15: note: candidate function not viable: no known conversion from 'std::__1::shared_ptr<arangodb::Thread>' to 'std::__1::weak_ptr<(anonymous namespace)::JobQueueThread>' for 1st argument
    weak_ptr& operator=(weak_ptr&& __r) _NOEXCEPT;
              ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/memory:4991:9: note: candidate template ignored: could not match 'weak_ptr' against 'shared_ptr'
        operator=(weak_ptr<_Yp> const& __r) _NOEXCEPT;
        ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/memory:5002:9: note: candidate template ignored: could not match 'weak_ptr' against 'shared_ptr'
        operator=(weak_ptr<_Yp>&& __r) _NOEXCEPT;
        ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/memory:5009:13: note: candidate template ignored: disabled by 'enable_if' [with _Yp = arangodb::Thread]
            is_convertible<_Yp*, element_type*>::value,
            ^
1 error generated.
make[2]: *** [arangod/CMakeFiles/arangod.dir/Scheduler/JobQueue.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [arangod/CMakeFiles/arangod.dir/all] Error 2
make: *** [all] Error 2
```

**Comment by baslr** [comment]
The `arangodb.rb` files doesnt set `CMAKE_OSX_DEPLOYMENT_TARGET`. Doxygen does it
https://github.com/Homebrew/homebrew-core/blob/master/Formula/doxygen.rb

**Comment by jsteemann** [comment]
@baslr: is this any relevant? I pushed a fix for the C++ code that triggered the error, and now it seems to compile fine. Do we still need to set the CMAKE_OSX_DEPLOYMENT_TARGET?

**Comment by baslr** [comment]
I guess yes. The CMakeLists.txt sets it to 10.11. Which is probably not a good idea if I compile on 10.10 and the 10.11 SDK is selected. Or I have the 10.12 SDK with improvements and the 10.11 SDK is chosen.

**Comment by jsteemann** [comment]
@baslr : got it. Who's going to fix it then?

**Comment by baslr** [comment]
I will create a pr in the arangodb-helper brew repo.

**Comment by jsteemann** [comment]
Thanks.

**Comment by baslr** [comment]
here is the pr https://github.com/arangodb-helper/homebrew-core/pull/1

**Comment by dothebart** [comment]
fixed with the release of 3.1.14
